pkg/network/tracer/ebpf_conntracker.go

```diff
@@ -172,25 +167,60 @@ func formatKey(netns uint32, tuple *ct.IPTuple) *ConnTuple {
 		return nil
 	}
 
-	return newConnTuple(0,
-		netns,
-		util.AddressFromNetIP(*tuple.Src),
-		util.AddressFromNetIP(*tuple.Dst),
-		*tuple.Proto.SrcPort,
-		*tuple.Proto.DstPort,
-		proto)
+	nct := &netebpf.ConntrackTuple{
+		Netns: netns,
+		Sport: *tuple.Proto.SrcPort,
+		Dport: *tuple.Proto.DstPort,
+	}
+	src := util.AddressFromNetIP(*tuple.Src)
+	nct.Saddr_l, nct.Saddr_h = util.ToLowHigh(src)
+	nct.Daddr_l, nct.Daddr_h = util.ToLowHigh(util.AddressFromNetIP(*tuple.Dst))
+
+	switch len(src.Bytes()) {
+	case net.IPv4len:
+		nct.Metadata |= uint32(netebpf.IPv4)
+	case net.IPv6len:
+		nct.Metadata |= uint32(netebpf.IPv6)
```

Comment:
I know it shouldn't be possible to hit this code path, but for safety against future changes it would be nice for this switch statement to have a `default` case which returns `nil`.

Reply:
Fixed